.github/workflows/go.yml

```
@@ -1,25 +1,26 @@
 name: Go
 
 on:
-  push:
-    branches: [ master ]
   pull_request:
-    branches: [ master ]
+    types: [ opened, synchronize, reopened, closed, edited ]
+    branches:
+      - master
 
 jobs:
 
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
 
-    - name: Set up Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: 1.16
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.22
```

Can we do 
```suggestion
          go-version: 'stable'
```

https://github.com/actions/setup-go?tab=readme-ov-file#using-stableoldstable-aliases

You may need to update to a later version of actions/setup-go